modules/services/authorizing/src/main/java/org/fao/geonet/authorizing/SecurityConfigurer.java

```diff
@@ -9,31 +9,87 @@
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerEndpointsConfigurer;
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerSecurityConfigurer;
+import org.springframework.security.oauth2.provider.token.DefaultTokenServices;
+import org.springframework.security.oauth2.provider.token.TokenStore;
+import org.springframework.security.oauth2.provider.token.store.JwtAccessTokenConverter;
+import org.springframework.security.oauth2.provider.token.store.JwtTokenStore;
+import java.security.KeyPair;
 
-@EnableWebSecurity
 @Configuration
-public class SecurityConfigurer extends WebSecurityConfigurerAdapter {
+@EnableAuthorizationServer
+public class SecurityConfigurer extends AuthorizationServerConfigurerAdapter {
 
-  @Autowired
-  public void configureGlobal(AuthenticationManagerBuilder auth)
-      throws Exception {
-    auth.inMemoryAuthentication().withUser("momo")
-        .password("{noop}password").roles("USER");
+  AuthenticationManager authenticationManager;
+  PasswordEncoder passwordEncoder;
+  UserDetailsService userDetailsService;
+
+  public SecurityConfigurer(AuthenticationConfiguration authenticationConfiguration, PasswordEncoder passwordEncoder, UserDetailsService userDetailsService) throws Exception {
+    this.authenticationManager = authenticationConfiguration.getAuthenticationManager();
+    this.passwordEncoder = passwordEncoder;
+    this.userDetailsService = userDetailsService;
   }
 
-  @Bean
-  public AuthenticationManager customAuthenticationManager() throws Exception {
-    return authenticationManager();
+  @Override
+  public void configure(AuthorizationServerSecurityConfigurer security) throws Exception {
+    security
+        .tokenKeyAccess("permitAll()")
+        .checkTokenAccess("isAuthenticated()")
+        .allowFormAuthenticationForClients();
+  }
+
+  @Override
+  public void configure(ClientDetailsServiceConfigurer clients) throws Exception {
+    clients
+        .inMemory() // (2)
+        .withClient("test-client")
+        .secret("{noop}noonewilleverguess"  ) // (3)
+        .scopes("any") // (4)
+        .autoApprove(true)
+        .authorizedGrantTypes("password", "refresh_token");
   }
 
   @Override
-  protected void configure(HttpSecurity http) throws Exception {
-    http.csrf().disable();
+  public void configure(AuthorizationServerEndpointsConfigurer endpoints) throws Exception {
+    endpoints.authenticationManager(this.authenticationManager)
+        .accessTokenConverter(accessTokenConverter())
+        .userDetailsService(userDetailsService)
+        .tokenStore(tokenStore());
+  }
+
+  @Bean
+  public TokenStore tokenStore() {
+    return new JwtTokenStore(accessTokenConverter());
+  }
+
+  @Bean
+  public JwtAccessTokenConverter accessTokenConverter() {
+    JwtAccessTokenConverter converter = new JwtAccessTokenConverter();
+    converter.setSigningKey("123");
```

Comment:
here to set https://github.com/geonetwork/geonetwork-microservices/blob/main/modules/support-services/configuring/src/main/resources/config/geonetwork-prod.properties#L1 or better encrypting scheme